pkg/api/config.go

```diff
@@ -42,10 +42,17 @@ func (cfg Config) GetGlobalDestinations(labels map[string]string) services.Desti
 		for _, trigger := range triggers {
 			if s.MatchesTrigger(trigger) && s.Selector.Matches(fields.Set(labels)) {
 				for _, recipient := range s.Recipients {
-					parts := strings.Split(recipient, ":")
-					dest := services.Destination{Service: parts[0]}
-					if len(parts) > 1 {
-						dest.Recipient = parts[1]
+					var before, after string
+					if i := strings.Index(recipient, ":"); i >= 0 {
```

Comment:
This PR supports sending Matrix messages to Matrix rooms and room aliases.

A Matrix room is identified with the syntax `!opaque_id:domain.tld` [[0]](https://spec.matrix.org/v1.5/appendices/#room-ids-and-event-ids) and a room alias with `#room_alias:domain.tld` [[1]](https://spec.matrix.org/v1.5/appendices/#room-aliases).

So the code change here in `pkg/api/config.go` is to make it so that the `domain.tld` part isn't lost.  Here's an example config; notice that the recipients have two `:`s in them: 
```yaml
notifications:
  notifiers:
    service.matrix: |
      accessToken: $matrix-token
      deviceID: $matrix-deviceid
      userID: $matrix-userid

  subscriptions:
    - recipients:
        - 'matrix:#my-room-alias:example.org'
        - 'matrix:!my-room-id:example.org'
      triggers:
        - on-deployed
        - on-health-degraded
        - on-sync-failed
        - on-sync-running
        - on-sync-status-unknown
        - on-sync-succeeded
```

Without this PR's change to `pkg/api/config.go`, we'd lose the `example.org` from the recipients and our Destinations for the Send function would be `#my-room-alias` and `!my-room-id` (which we can't send notifications to).

Reply:
What I'm hoping is that once this project upgrades to golang 1.18, we can revert this PR's 2nd commit to simplify the code a bit.